File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    k_pack = []
    
    for i in range(len(score)):
        k_pack = sorted(score[0:i+1], reverse=True)
        k_pack = k_pack[0:i+1] if i<k else k_pack[0:k]
        answer.append(min(k_pack))
    
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.Arrays;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        int[] top = new int[k];
        int min = 2001;
        int j = 0;
        
        for(int i=0; i<score.length; i++) {
            if(i<k) {
                top[i] = score[i];
                if(min > score[i])
                    min = score[i];
                answer[j] = min;
            } else {
                Arrays.sort(top);
                if(top[0] < score[i])
                    top[0] = score[i];
                Arrays.sort(top);
                answer[j] = top[0];
            }
            j++;
        }
        return answer;
    }
}